libs/utils/android/workloads/viewer.py

```diff
@@ -0,0 +1,108 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2015, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class Viewer(Workload):
+    """
+    Android generic Viewer workload
+    """
+    
+    # Let the system pick the best package
+    package = ''
+
+    def __init__(self, test_env):
+        super(Viewer, self).__init__(test_env)
+        self._log = logging.getLogger('Viewer')
+        self._log.debug('Workload created')
+
+        # Set of output data reported by the browser
+        self.db_file = None
+
+    def run(self, out_dir, uri, workload, collect=''):
+        """
+        Run viewer workload
+
+        :param out_dir: Path to experiment directory where to store results.
+        :type out_dir: str
+
+        :param uri: The URI of the item to display
+        :type location_search: str
+
+        :param workload: Function pointer to the workload to run once item
+        is opened and being viewed.
+        :type workload: func
```

Comment:
Is not a function pointer just like a pure virtual method? I would probably better go for a subclass me too.
It could be over-killing for some simple use-cases... but if we need to integrate more complex use-cases in the future we already have the infrastructure in place. Isnt' it?